[20.01] Add requirements, fix legacy tools and move them out of GALAXY_LIB_TOOLS_UNVERSIONED

mvdbeek opened the pull request:
~~Needs a new galaxy-sequence-utils release~~

mvdbeek committed:
Fix vcf_to_maf_customtrack

jdavcs commented on tools/maf/vcf_to_maf_customtrack.xml:
```diff
@@ -1,5 +1,9 @@
+<tool id="vcf_to_maf_customtrack1" name="VCF to MAF Custom Track" version="1.0.1">
   <description>for display at UCSC</description>
+  <requirements>
+    <requirement type="package" version="1.1.3">galaxy_sequence_utils</requirement>
```
1.1.4?

mvdbeek committed:
Drop __next__

mvdbeek committed:
Bump version and add profile

mvdbeek committed:
Move secure_hash_message_digest to  GALAXY_LIB_TOOLS_VERSIONED

And give it its own dependency.

jdavcs commented on tools/maf/vcf_to_maf_customtrack.py:
```diff
@@ -13,51 +13,43 @@
 class PopulationVCFParser(Iterator):
     def __init__(self, reader, name):
-        self.reader = reader
+        self.reader = iter(reader)
```
A tangent. I'm wondering... We have to do this because the reader's `__iter__` method does not return `self` (i.e., it's not an iterator without that), right? Would it makes sense to refactor that code so that it doesn't require this step (wrapping in `iter()` and, thus, is more straightforward to use?

mvdbeek replied:
Sure, why not, but I don't think that would be a good use of time. These tools are legacy and I doubt anyone would really build new stuff based on galaxy-sequence-utils.

jdavcs replied:
Oh OK, makes sense. Never mind then, there's only so much time.

mvdbeek committed:
Make join1 use proper requirements, move to GALAXY_LIB_TOOLS_VERSIONED

mvdbeek committed:
Move wiggle_to_interval out of GALAXY_LIB_TOOLS_UNVERSIONED

mvdbeek committed:
Move aggregate_scores_in_intervals2 out of GALAXY_LIB_TOOLS_UNVERSIONED

mvdbeek committed:
Use text mode in maf_utilites, set non-zero exit code

I don't see any binary formats being handled here, but strings are being
written to file handles in binary mode, so this should be correct.

Would be great to move maf_utilities out so that tools can depend
on this explicitly.

mvdbeek committed:
Move CONVERTER_fastq_to_fqtoc0 out of GALAXY_LIB_TOOLS_UNVERSIONED

mvdbeek committed:
Move CONVERTER_tar_to_directory out of GALAXY_LIB_TOOLS_UNVERSIONED

mvdbeek commented:
I think that's all we can do for Converters and tools shipped with galaxy. It would be good to move maf_utilties and gff_utils to a separate package (or galaxy-sequence-utils), with that in place I think only `upload1`, `__DATA_FETCH__` and `send_to_cloud` would need to remain on the list (and maybe we could even make them depend on a fatter set of galaxy dependencies).
